universe/ShipDesign.cpp

```diff
@@ -603,7 +607,7 @@ HullType::~HullType()
 
 void HullType::Init(std::vector<std::unique_ptr<Effect::EffectsGroup>>&& effects) {
     if (m_fuel != 0)
-        m_effects.push_back(IncreaseMeter(METER_MAX_FUEL,       m_fuel));
+        m_effects.push_back(IncreaseMeter(METER_MAX_FUEL,       m_fuel, 500)); // FIXME MAGIC NUMBER
```

Comment:
I think this should actually be done in FOCS rather than C++ code. This bakes a specific priority into the C++, which otherwise isn't done (?) or at least isn't done very much. The relative values of the priorities are all set from the values in scripts, mainly in priorities.macros, so putting a value into the C++ that is determined mainbly based on what those scripts contain seems like a bad idea.

So in that case, I'd disable to autogenerated effect by adding `NoDefaultCapacityEffect` to the part scripts, and instead add a custom scripted fuel effect to the hulls with whatever priority is desired.

This probably needs to be a new [[VERY_VERY_LATE_PRIORITY]] or similar, since the fuel effeciency scaling effects are currently [[VERY_LATE_PRIORITY]]... though perhaps all the fuel effects could be bumped down one priority step so the unscaled hull contribution could be put a [[VERY_LATE_PRIOIRITY]].

Reply:
Given all that, I'd push this to post-release. It sound like the current issue this is intended to fix is cosmetic, regarding ordering in effect accounting?

Reply:
I would rather have the priority be a functional value in priorities.macros and slurp that into c++
Adding language features to selectively disable parts of default effects sound like a lot of headaches (and much work as well).

> Given all that, I'd push this to post-release. It sound like the current issue this is intended to fix is cosmetic, regarding ordering in effect accounting?

You can not really understand the effect from looking at the effect values I think; but maybe the current state is acceptable. Have a look and decide:
![201910__2620__negative_fuel_effects_accounting__great_looks_bad](https://user-images.githubusercontent.com/3252041/68550854-3e9d4d80-0407-11ea-8ac7-e1bd0f9b8540.png)

Intended look:
![201910__2620__negative_fuel_effects_accounting__reordered_effects](https://user-images.githubusercontent.com/3252041/68550961-15c98800-0408-11ea-9f93-0bd5c79c2a59.png)

Reply:
> I would rather have the priority be a functional value in priorities.macros and slurp that into c++

I don't know what that means.

> Have a look and decide:

The effect accounting doesn't make sense / isn't clear regardless of the ordering of the list, I think.

Reply:
> > I would rather have the priority be a functional value in priorities.macros and slurp that into c++
> 
I meant we have e.g. FUNCTIONAL_* stringtable entries using the stringtable mechanism to use values from FOCS in C++. We can read a priority defined in priorities.macros e.g. AFTER_FUEL_EFFICIENCY_PRIORITY or APPLY_HULL_FUEL_PRIORITY = 500
Then we do not have the magic value in the code.


> The effect accounting doesn't make sense / isn't clear regardless of the ordering of the list, I think.

The point here is the scaling also for positive values. I expect that the fuel efficiency 400% is applied to the effects above because that is how it usually works. And with the intended UI that expectation is matched.